core/types/transaction.go

```diff
@@ -204,31 +204,30 @@ func (tx *Transaction) DecodeRLP(s *rlp.Stream) error {
 
 // MarshalJSON encodes the web3 RPC transaction format.
 func (tx *Transaction) MarshalJSON() ([]byte, error) {
-	return TransactionMarshalJSON(tx)
+	return tx.data.TransactionMarshalJSON()
 }
 
 // UnmarshalJSON decodes the web3 RPC transaction format.
 func (tx *Transaction) UnmarshalJSON(input []byte) error {
-	dec, err := TransactionUnmarshalJSON(input)
+	err := tx.data.TransactionUnmarshalJSON(input)
```

Comment:
Why were we able to remove `dec`?

Reply:
I re-auto generated the JSON marshaling code which eliminated manual edits to the file, removing the tx meta from the JSON serialization. It is handled correctly in the RPC